### ISSUE TITLE
InteliSense Doesnt Parse Laravel Code

### ISSUE THREAD

**Issue by CowboyDiplomacy** [HvyIndustries/crane#66]
Intellisense does not in any way seem to parse multiple files, making coding using popular PHP frameworks such as Laravel useless in VS Code.

**Comment by TheColorRed** [comment]
There is an issue in the current release that breaks the parser from parsing Laravel projects. The next release has some fixes for this.

**Comment by nevadascout** [comment]
@TheColorRed Are you able to verify if the 0.2 release candidate is able to parse Laravel projects successfully?

**Comment by TheColorRed** [comment]
@nevadascout it is working much better, however, when building the tree we are still getting some errors:

```
[ERROR] There was a problem parsing PHP file: c:\Users\rnaddy\Documents\www\gamesmart\vendor\phpunit\phpunit\tests\Regression\GitHub\797\bootstrap797.php
[ERROR] [TypeError: Cannot read property '0' of undefined]
```

Here is the file contents:

```php
<?php
// If process isolation fails to include this file, then
// PHPUnit_Framework_TestCase itself does not exist. :-)
require __DIR__ . '/../../../bootstrap.php';

const GITHUB_ISSUE = 797;
```
<hr>

```
[ERROR] There was a problem parsing PHP file: c:\Users\rnaddy\Documents\www\gamesmart\vendor\phpspec\phpspec\features\bootstrap\Matcher\ValidJUnitXmlMatcher.php
[ERROR] [TypeError: Cannot read property '0' of undefined]
```

```php
<?php

namespace Matcher;

use PhpSpec\Exception\Example\FailureException;
use PhpSpec\Matcher\MatcherInterface;
use Symfony\Component\Console\Tester\ApplicationTester;

const JUNIT_XSD_PATH = '/src/PhpSpec/Resources/schema/junit.xsd';

class ValidJUnitXmlMatcher implements MatcherInterface
{
    /**
     * Checks if matcher supports provided subject and matcher name.
     *
     * @param string $name
     * @param mixed $subject
     * @param array $arguments
     *
     * @return Boolean
     */
    public function supports($name, $subject, array $arguments)
    {
        return ($name == 'haveOutputValidJunitXml' && $subject instanceof ApplicationTester);
    }

    /**
     * Evaluates positive match.
     *
     * @param string $name
     * @param mixed $subject
     * @param array $arguments
     */
    public function positiveMatch($name, $subject, array $arguments)
    {
        $dom = new \DOMDocument();
        $dom->loadXML($subject->getDisplay());
        if (!$dom->schemaValidate(__DIR__ . '/../../..' . JUNIT_XSD_PATH)) {
            throw new FailureException(sprintf(
               "Output was not valid JUnit XML"
            ));
        }
    }

    /**
     * Evaluates negative match.
     *
     * @param string $name
     * @param mixed $subject
     * @param array $arguments
     */
    public function negativeMatch($name, $subject, array $arguments)
    {
        throw new FailureException('Negative JUnit matcher not implemented');
    }

    /**
     * Returns matcher priority.
     *
     * @return integer
     */
    public function getPriority()
    {
        return 51;
    }
}
```

```
[ERROR] There was a problem parsing PHP file: c:\Users\rnaddy\Documents\www\gamesmart\vendor\phpunit\php-token-stream\tests\_fixture\class_with_method_that_declares_anonymous_class.php
[ERROR] [ReferenceError: T_STRING is not defined]
[ERROR] There was a problem parsing PHP file: c:\Users\rnaddy\Documents\www\gamesmart\vendor\phpunit\php-token-stream\tests\_fixture\class_with_method_that_declares_anonymous_class2.php
[ERROR] [ReferenceError: T_STRING is not defined]
```

There are a few others, and the only thing that I can see that they had in common, was that they used `__DIR__` and/or `const`.

Other than that, I was able to get completion items for most things. I was not able to get completion items for static items, such as `Route::` which would have suggested `group`, `get`, `post`, etc.

**Comment by nevadascout** [comment]
Thanks, I'll look into this later!

Was it just that one class, or are all static items broken?

**Comment by TheColorRed** [comment]
It might be that one class, `Request` and `Auth` seemed to work.

**Comment by nevadascout** [comment]
So the parse issue is related to constants, that should be fixed in 7f3be73

**Comment by nevadascout** [comment]
This will be resolved in the 0.2 release.

If any further issues arise, please open a new issue!
